caproto/threading/client.py

```diff
@@ -1455,6 +1455,7 @@ def read(self, *, wait=True, callback=None, timeout=2, data_type=None,
         deadline = time.monotonic() + timeout if timeout is not None else None
         ioid_info['deadline'] = deadline
         self.circuit_manager.send(command)
+        self.log.debug(f"{self.name!r}: {command}")
```

Comment:
While I really like f-strings, for logging there's no better choice than the old-style string formatting. This will cause command repr to be generated even if the log level is not set such that this message would go through.

Reply:
:man_facepalming: Knew that, literally opened a terminal to play with the details of that implementation yesterday, and then did not apply it....